### ISSUE TITLE
"accumulates inputs until the target value is matched"

### ISSUE THREAD

**Issue by PeterGarner** [bitcoinjs/coinselect#34]
if the target value is 'matched', where do the fees come from?

**Comment by dcousens** [comment]
From the inputs?
The `fee = input - output`,  where `input >= output`.

I'll update the README example to be copy-pastable.

**Comment by PeterGarner** [comment]
I think what’s confusing is the term matched. If input == (matches) output then fee = 0. Do you mean ‘reached’?

**Comment by dcousens** [comment]
Yes, reached.

PR accepted.

**Comment by dcousens** [comment]
It does say `reached`
https://github.com/bitcoinjs/coinselect/blame/11e0e925ccc939c2b0889b43020a590e89a66b20/README.md#L15

If you are referring to `blackjack`:
https://github.com/bitcoinjs/coinselect/blob/11e0e925ccc939c2b0889b43020a590e89a66b20/README.md#L16

Then,  it is `matched`.
https://github.com/bitcoinjs/coinselect/blob/11e0e925ccc939c2b0889b43020a590e89a66b20/blackjack.js#L22-L33

Where target value is `outAccum + fee`.

**Comment by PeterGarner** [comment]
I'm confusing myself.  Of course, 'matched' is the correct term for Blackjack (like the game), not reached.  

What I should have focussed on was the term 'target value'.  This refers to the output + pre-calculated fee, is that right?   

```var fee = feeRate * (bytesAccum + inputBytes)```

Whereas I was reading 'target value' == THE output value, as in 
```let **targets** = [{ address: payToAddr, **value**: amount }]```

**Comment by dcousens** [comment]
> What I should have focussed on was the term 'target value'. This refers to the output + pre-calculated fee, is that right?

Yes,  the "target value" refers to `outAccum + fee`,  and is variable in that it increases as the `fee` increases (as inputs are added).

**Comment by dcousens** [comment]
Do you have another suggestion for the term?

You are right in that it is ambiguous and could be better.

**Comment by PeterGarner** [comment]
I think 'target value' is fine if referring to the actual amount you want the target address to receive and fits with the terms already used when constructing the 'targets' array.     I would only add '+ fees' and leave it for people to read the code to see how the fees are dynamically determined and included.
